src/scraper.rs

```diff
@@ -130,6 +131,55 @@ pub struct Days {
     pub sunday: bool,
 }
 
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum Day {
+    Monday,
+    Tuesday,
+    Wednesday,
+    Thursday,
+    Friday,
+    Saturday,
+    Sunday,
+}
+
+impl Day {
+    pub const ALL: [Self; 7] = [
+        Self::Monday,
+        Self::Tuesday,
+        Self::Wednesday,
+        Self::Thursday,
+        Self::Friday,
+        Self::Saturday,
+        Self::Sunday,
+    ];
+
+    pub const WEEKDAYS: [Self; 5] = [
+        Self::Monday,
+        Self::Tuesday,
+        Self::Wednesday,
+        Self::Thursday,
+        Self::Friday,
+    ];
+
+    pub(crate) fn is_in_days(&self, days: Days) -> bool {
+        match self {
+            Day::Monday => days.monday,
+            Day::Tuesday => days.tuesday,
+            Day::Wednesday => days.wednesday,
+            Day::Thursday => days.thursday,
+            Day::Friday => days.friday,
+            Day::Saturday => days.saturday,
+            Day::Sunday => days.sunday,
+        }
+    }
+}
+
+impl fmt::Display for Day {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
```

Comment:
should the `to_lowercase()` call occur in here? what if we want the PascalCase'd day name at some point in the future? or to serialize it